### DISCUSSION TITLE
[Q&A] - Error upgrade td-agent v.4 to fluent-package v5

### DISCUSSION THREAD

**Discussion by raffleo24** [fluent/fluentd#4701] [Q&A]
### What is a problem?

Hi, we upgraded td agent v.4 to fluent-package v.5.1.0 on a Windows 2016 system.
The package was downloaded from here: https://td-agent-package-browser.herokuapp.com/5/windows.
After performing the various steps as indicated in the guide, we see that previous configuration takes it correctly and also we copied the certificate folder to the new fluent folder.
However, it seems not to work properly anymore (it does not send logs anymore). 
From the fluent.log file appear some WARN (already present before the upgrade) but one log in particular was not present before: [warn]: #0 failed to flush the buffer. retry_times=2 next_retry_time=2024-11-07 10:12:14 +0100 chunk="xx" error_class=OpenSSL::SSL::SSLError error="SSL_connect returned=1 errno=0 peeraddr=10.11.12.xxx:3535 state=error: certificate verify failed (CA certificate key too weak)".

Instead, always to perform a version upgrade from 4 to 5 but on a linux system, it gives us the following error after running the rpm package:
rpm -ivh fluent-package-5.0.0-1.el8.x86_64.rpm error: Failed dependencies: td-agent < 5.0.0 is obsolete by fluent-package-5.0.0-1.el8.x86_64

### Describe the logs of Fluentd

[warn]: #0 failed to flush the buffer. retry_times=2 next_retry_time=2024-11-07 10:12:14 +0100 chunk="xxx" error_class=OpenSSL::SSL::SSLError error="SSL_connect returned=1 errno=0 peeraddr=10.11.12.xxx:3535 state=error: certificate verify failed (CA certificate key too weak)"

 rpm -ivh fluent-package-5.0.0-1.el8.x86_64.rpm
error: Failed dependencies:
        td-agent < 5.0.0 is obsoleted by fluent-package-5.0.0-1.el8.x86_64

### Environment

```markdown
- Fluentd version: 1.17
- TD Agent version: v4
- Fluent Package version: 5.1.0
- Docker image (tag):
- Operating system: Windows
- Kernel version:
```

**Comment by ashie** [comment]
> [warn]: #0 failed to flush the buffer. retry_times=2 next_retry_time=2024-11-07 10:12:14 +0100 chunk="xx" error_class=OpenSSL::SSL::SSLError error="SSL_connect returned=1 errno=0 peeraddr=10.11.12.xxx:3535 state=error: certificate verify failed (CA certificate key too weak)".

It seems concerned with bundled OpenSSL version.
fluent-package 5.0 upgrades bundled OpenSSL to 3.x while td-agent v4 bundles OpenSSL 1.1.1.
By upgrading OpenSSL, perhaps some security policy has been tightened so that your old certificate is rejected.
You might need to upgrade your certificate or relax security settings.

> rpm -ivh fluent-package-5.0.0-1.el8.x86_64.rpm
> error: Failed dependencies:
> td-agent < 5.0.0 is obsoleted by fluent-package-5.0.0-1.el8.x86_64

You need to use `rpm -Uvh` instead of `rpm -ivh` to upgrade it.
